src/EditorFeatures/Core.Wpf/InlineErrors/InlineErrorTag.cs

```diff
@@ -0,0 +1,153 @@
+                    Margin = new Thickness(1, 0, 5, 0)
+                };
+
+                var statusImage = new InlineUIContainer(image);
+                block.Inlines.Add(statusImage);
+            }
+
+            link.RequestNavigate += HandleRequestNavigate;
+            block.Inlines.Add(link);
+            block.Inlines.Add(": " + _diagnostic.Message);
+            block.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            block.Arrange(new Rect(block.DesiredSize));
+
+            var border = new Border
+            {
+                BorderBrush = format.BackgroundBrush,
+                BorderThickness = new Thickness(1),
+                Background = Brushes.Transparent,
+                Child = block,
+                CornerRadius = new CornerRadius(2),
+                // Highlighting lines are 2px buffer. So shift us up by one from the bottom so we feel centered between them.
+                Margin = new Thickness(10, top: 0, 0, bottom: 0),
+                Padding = new Thickness(1)
+            };
+
+            border.Measure(new Size(double.PositiveInfinity, double.PositiveInfinity));
+            // Need to set these properties to avoid unnecessary reformatting because some dependancy properties
+            // affect layout.
+            // TODO: Not sure if these are needed anymore since the errors are not intratextadornment tags
+            TextOptions.SetTextFormattingMode(border, TextOptions.GetTextFormattingMode(view.VisualElement));
+            TextOptions.SetTextHintingMode(border, TextOptions.GetTextHintingMode(view.VisualElement));
+            TextOptions.SetTextRenderingMode(border, TextOptions.GetTextRenderingMode(view.VisualElement));
+
+            view.ViewportWidthChanged += ViewportWidthChangedHandler;
+
+            return new GraphicsResult(border,
+                    () => view.ViewportWidthChanged -= ViewportWidthChangedHandler);
+
+            void ViewportWidthChangedHandler(object s, EventArgs e)
+            {
+                if (Location is InlineErrorsLocations.HookedToWindow)
+                {
+                    Canvas.SetLeft(border, view.ViewportWidth - border.DesiredSize.Width);
+                }
+            }
+        }
+
+        private ImageMoniker? GetMoniker()
+        {
+            switch (_diagnostic.Severity)
+            {
+                case DiagnosticSeverity.Warning:
+                    return KnownMonikers.StatusWarning;
+                case DiagnosticSeverity.Error:
+                    return KnownMonikers.StatusError;
+                default:
+                    return null;
+            }
+        }
+
+        /// <summary>
+        /// Navigates to the requrest URL
+        /// </summary>
+        private void HandleRequestNavigate(object sender, RoutedEventArgs e)
+        {
+            var link = (Hyperlink)sender;
+            var uri = link.NavigateUri.ToString();
+            Process.Start(uri);
```

Comment:
@mavasani is this the right way to navigate to a uri?

Reply:
https://github.com/dotnet/roslyn/blob/315c2e149ba7889b0937d872274c33fcbfe9af5f/src/VisualStudio/Core/Impl/Options/CodeStyleNoticeTextBlock.xaml.cs#L32